### ISSUE TITLE
Test on a scanned point cloud of a real architectual scene?

### ISSUE THREAD

**Issue by Tomoki-0526** [yilinliu77/NVDNet#2]
Hello, I wonder if your work is valid on a scanned architectual cloud. I run "generate_test_voronoi_data.py" file to generate the mesh but find the output very strange.
Here is the origin cloud:
![屏幕截图 2024-07-13 221630](https://github.com/user-attachments/assets/2168a34d-7771-4bb8-add2-c07602a7723f)
but the output is like:
![屏幕截图 2024-07-13 221910](https://github.com/user-attachments/assets/6bd1cb27-b94f-460d-b521-6c3b24916c5d)
So what can i do to preprocess the data?

**Comment by yilinliu77** [comment]
Hi, thanks for your interest in our work. For large-scale data, you might want to split it into small patches and run the inference code in parallel. However, the current model has not been trained on noise data. If the data contains serious noise, it is possible to predict an incomplete voronoi diagram and affect the subsequent post-processing pipeline.

**Comment by Silvarebacca** [comment]
@yilinliu77 Hi, if we train the model on noisy data (noisy or even incomplete), do you think it could be possible to predict the complete voronoic diagram and the subsequent watertight mesh? Or any advice for dealing with noisy data?  Thank U.

**Comment by Silvarebacca** [comment]
@yilinliu77 I test the pre-trained model on real-world point clouds which contain some noise but not so serious. The final mesh results with a color map are not very good. Specifically, as shown in the figure, the surface that should be a flat, planar area is represented using a variety of colors, rather than being a uniform, single color. In addition to training the model on noisy data, whatelse should I do to improve the performance?

![Image](https://github.com/user-attachments/assets/d261ba2a-f93b-4290-8aef-3ccdb41a5a14)

**Comment by yilinliu77** [comment]
For a noisy point cloud, we rely on Neural Dual Contouring to get a cleaner version of the data. You could also try it before feeding the data into NVD-Net for the segmentation.
